samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -0,0 +1,164 @@
+
+	bool result = PersistentStorage::Instance().Load(node, &tempData, sizeof(T), readSize);
+
+	data = tempData;
+	return result;
+}
+
+PersistentStorageNode CreateIndexNode(uint8_t bridgedDeviceIndex, PersistentStorageNode *parent)
+{
+	char index[BridgeStorageManager::kMaxIndexLength + 1] = { 0 };
+
+	snprintf(index, sizeof(index), "%d", bridgedDeviceIndex);
+
+	return PersistentStorageNode(index, strlen(index), parent);
+}
+
+} /* namespace */
+
+bool BridgeStorageManager::StoreBridgedDevicesCount(uint8_t count)
+{
+	return PersistentStorage::Instance().Store(&mBridgedDevicesCount, &count, sizeof(count));
+}
+
+bool BridgeStorageManager::LoadBridgedDevicesCount(uint8_t &count)
+{
+	return LoadSimpleData(&mBridgedDevicesCount, count);
+}
+
+bool BridgeStorageManager::StoreBridgedDevicesIndexes(uint8_t *indexes, uint8_t count)
+{
+	if (!indexes) {
+		return false;
+	}
+
+	return PersistentStorage::Instance().Store(&mBridgedDevicesIndexes, indexes, count);
+}
+
+bool BridgeStorageManager::LoadBridgedDevicesIndexes(uint8_t *indexes, uint8_t maxCount, size_t &count)
+{
+	if (!indexes) {
+		return false;
+	}
+
+	return PersistentStorage::Instance().Load(&mBridgedDevicesIndexes, indexes, maxCount, count);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceEndpointId(uint16_t endpointId, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceEndpointId);
+
+	return PersistentStorage::Instance().Store(&id, &endpointId, sizeof(endpointId));
+}
+
+bool BridgeStorageManager::LoadBridgedDeviceEndpointId(uint16_t &endpointId, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceEndpointId);
+
+	return LoadSimpleData(&id, endpointId);
+}
+
+bool BridgeStorageManager::RemoveBridgedDeviceEndpointId(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceEndpointId);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceNodeLabel(const char *label, size_t labelLength,
+						       uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Store(&id, label, labelLength);
+}
+
+bool BridgeStorageManager::LoadBridgedDeviceNodeLabel(char *label, size_t labelMaxLength, size_t &labelLength,
+						      uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Load(&id, label, labelMaxLength, labelLength);
+}
+
+bool BridgeStorageManager::RemoveBridgedDeviceNodeLabel(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceType(uint16_t deviceType, uint8_t bridgedDeviceIndex)
```

Comment:
NIT: Can the function names be shortened? for example from `StoreBridgedDeviceType` to `StoreType`? Or do you think it will be too ambiguous then?

Reply:
I don't have strong opinion on that. As you wrote, I named it like this to avoid ambiguity and clearly inform that it stores or loads some parameter of a single bridged device. However if you, as a reviewer think it would not be ambiguous to not have `BridgedDevice` in every parameter name we can skip it.

Reply:
To keep the C++ style we can also add `namespace BridgedDevice` and then use `BridgeStorageManager::BridgedDevice::StoreType()`. But I also don't have a strong opinion on that, only these long names caught my eye.